src/test/test-seccomp.c

```diff
@@ -949,24 +949,25 @@ TEST(lock_personality) {
                 return;
         }
 
-        assert_se(opinionated_personality(&current) >= 0);
-        /* On ppc64le sanitizers disable ASLR (i.e. by setting ADDR_NO_RANDOMIZE),
-         * which opinionated_personality() doesn't return. Let's tweak the current
-         * personality ourselves in such cases.
-         * See: https://github.com/llvm/llvm-project/commit/78f7a6eaa601bfdd6ae70ffd3da2254c21ff77f9
-         */
-        if (FLAGS_SET(safe_personality(PERSONALITY_INVALID), ADDR_NO_RANDOMIZE))
-                current |= ADDR_NO_RANDOMIZE;
+        assert_se(opinionated_personality(&current_opinionated) >= 0);
 
-        log_info("current personality=0x%lX", current);
+        log_info("current personality=0x%lX", (unsigned long)safe_personality(PERSONALITY_INVALID));
+        log_info("current opinionated personality=0x%lX", current_opinionated);
 
         pid = fork();
         assert_se(pid >= 0);
 
         if (pid == 0) {
-                assert_se(seccomp_lock_personality(current) >= 0);
+                unsigned long current;
 
-                assert_se((unsigned long) safe_personality(current) == current);
+                assert_se(seccomp_lock_personality(current_opinionated) >= 0);
+
+                /* Mask the top three bytes from the value that personality() returns, since they are used
+                 * for flags that emulate certain historical or architectural quirks and, most importantly,
+                 * opinionated_personality() doesn't return them, causing spurious test fails on some
+                 * "exotic" architectures/configurations. */
+                current = safe_personality(current_opinionated);
+                assert_se((current & 0xFF) == current_opinionated);
```

Comment:
hmm, opinionated_personality() currently uses as mask 0xffff. we probably should do that too here.

the definitions in /usr/include/sys/personality.h are so weird about this. A comment says "byte", but then the enum is carefully defined to always use 16bit 0x00?? values. 

I'd probably stick to 0xFFFFLU for this, looking at the definitions this sounds to be the most accurate. Dunno.

If we start making using of this mask at three places though, let's add a macro for this:

```c
#define OPINIONATED_PERSONALITY_MASK 0xFFFFUL
```

or so. and use it all three places.

Reply:
Yeah, I was confused about that too, but the man page also states that only the least significant byte is the actual personality:

>  The execution domain is a 32-bit  value
>       in  which  the  top  three bytes are set aside for flags that cause the kernel to modify the behavior of certain system
>       calls so as to emulate historical or architectural quirks.  The least significant byte is a value defining the  person‐
>      ality the kernel should assume.

(taken from `personality(2)`)